open-company/open-company-web
# Invite link

Card: https://trello.com/c/VsS8yYh2

Review with https://github.com/open-company/open-company-auth/pull/99

Test invite link:
- with an admin open Invite people from the navigation sidebar
- pick Invite via email
- click on Generate an invite link
- copy the link
- open the link in incognito
- [x] could you signup? did you land in the right team? did it collect your email, password and name?
- now with the admin check that the newly signed up user is a contributor
- now with the admin open again invite via email
- copy the invite link
- deactivate the link
- open the link in incognito
- [x] do you get an error?

Test email domains:
- with the admin go to Admin settings
- add `thisisatest.com` as email domain
- generate an invite via email link
- open the link in incognito
- signup using a `thisisatest.com` email
- [x] did it let you in? can you see AP?
- now open a new incognito window
- signup for a new org via email
- add `carrot.io` as email domain
- open in incognito the invite link you used before (third step of this test)
- signup using a `carrot.io` email
- [x] did you get an email wall asking to confirm your email address before we let you in?

Test Slack link:
- signup with email
- go to Invite people
- [x] do you see Invite via Slack greyed out and Add bot besides it clickable?
- click add Slack
- signup with a Slack org (better if you don't have it in the DB yet)
- [x] did it send you back to Invite via Slack panel?
- [x] do you see the invite via Slack link available?
- [x] do you see "Enable the Carrot bot for Slack" button?
- click it
- grant bot permissions
- [x] did it bring you back to Invite via Slack?
- [x] can you invite single users via Slack now?

Test invite regressions:
- go to Invite via email
- invite an Admin, a Contributor, and a Viewer
- accept all 3 invites
- [x] all users have the supposed permissions?
- go to Invite via Slack
- invite someone with a random user type
- login to Slack with that user and check he received the invitation link
- click the link and signup
